src/python/dart/service/email.py

```diff
@@ -43,21 +44,52 @@ def send_error_email(self, subject, body, to=None):
             to = self._cc_on_error
         self.send_email(subject, body, to, cc=cc)
 
+    def extract_action_messages(self, action, datastore):
+        values = dict(action_id=action.id,
+                      action_type_name=action.data.action_type_name,
+                      datastore_id=datastore.id,
+                      datastore_name=datastore.data.name,
+                      entity_link='',
+                      action_err_msg=action.data.error_message,
+                      action_ecs_task_arn=action.data.ecs_task_arn,
+                      workflow_id=action.data.workflow_id,
+                      workflow_instance_id=action.data.workflow_instance_id,
+                      engine_name=action.data.engine_name)
+
+        subject = '{action_status} Dart: action (action_id={action_id}, action_type_name={action_type_name})'
+        message = """action (action_id={action_id}, action_type_name={action_type_name}) {action_status}
+                     for datastore (datastore_id=%s, datastore_name=%s)
+                     {entity_link}
+
+                     action_err_msg={action_err_msg}
+
+                     workflow_id={workflow_id}, workflow_instance_id={workflow_instance_id}
+                     task-arn={action_ecs_task_arn}, engine={engine_name}"""
+
+        return (values, subject, message)
+
+
     def send_action_failed_email(self, action, datastore):
-        values = (action.id, action.data.action_type_name, datastore.id, datastore.data.name,
-                  self.get_entity_link('actions', action.id), action.data.error_message)
+        values, subject, message = self.extract_action_messages(action, datastore)
+
+        values.update({'action_status': 'FAILED'})
+        values.update({'entity_link': self.get_entity_link('actions', action.id)})
+
         self.send_error_email(
-            'FAILED Dart: action (id=%s, name=%s)' % (action.id, action.data.action_type_name),
-            'action (id=%s, name=%s) FAILED for datastore (id=%s, name=%s)\n\n%s\n\n%s' % values,
+            subject.format(**values),
+            dedent(message.format(**values)),
```

Comment:
Is this actually taking care of the indent? The first line isn't indented so there's no 'common' indentation to remove.